src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,140 @@
+                        removeInstructions(index, 2)
+
+                        // Remove the instruction that uses the resultRegister as an array reference to prevent an error.
+                        removeInstructions(index - 2, 1)
+                    }
+
+                    break
+                }
+            }
+
+            // Instruction that sets free register to "this", needed to restore the original value of the register.
+            var instructionToGetThis: TwoRegisterInstruction? = null
+
+            // Save user nickname to free register.
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodReference = (instruction as ReferenceInstruction).reference.toString()
+                    if (!methodReference.endsWith("Ljava/lang/String;")) continue
+
+                    // Get instruction that sets free register to "this".
+                    instructionToGetThis = mutableMethod.getInstruction<TwoRegisterInstruction>(index - 1)
+                    mutableMethod.addInstruction(
+                        index + 2,
+                        "move-object v${instructionToGetThis.registerA}, v${instructionToGetThis.registerA}"
+                    )
+                    break
+                }
+            }
+
+            if (instructionToGetThis == null) throw PatchException("Instruction to get \"this\" not found")
+
+            // Call the patched method and save the result to resultRegister.
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Ljava/lang/Long;") continue
+
+                    // Get the registers for tweet ID number (64 bits).
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+                    val sourcePlusOne = sourceRegister + 1
+
+                    // Call the patched method with the tweet ID and username and save the result to resultRegister.
+                    this.mutableMethod.addInstructions(
+                        index + 2,
+                        """
+                            invoke-static { v$sourceRegister, v$sourcePlusOne, v${instructionToGetThis.registerA} }, ${linkBuilderResult.method}
+                            move-result-object v$resultRegister
+                        """
+                    )
+
+                    // Restore the register that was used to store our string by duplicating the instruction that got "this".
+                    this.mutableMethod.addInstruction(index+4, instructionToGetThis as BuilderInstruction)
+                    break
+                }
+            }
```

Comment:
Without reading the app's source code, it is impossible to understand whats going on here. If possible, simplify the code